src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers/ApiControllerAnalyzer.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using cs = Microsoft.CodeAnalysis.CSharp;
+using vb = Microsoft.CodeAnalysis.VisualBasic;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers
+{
+    [ApplicableComponents(ProjectComponents.AspNetCore)]
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public class ApiControllerAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "UA0013";
+        private const string Category = "Upgrade";
+
+        public const string BadNamespace = "System.Web.Http";
+        public const string BadClassName = "ApiController";
+
+        private static readonly LocalizableString Title = new LocalizableResourceString(nameof(Resources.ApiControllerDescription), Resources.ResourceManager, typeof(Resources));
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(nameof(Resources.ApiControllerDescription), Resources.ResourceManager, typeof(Resources));
+        private static readonly LocalizableString Description = new LocalizableResourceString(nameof(Resources.ApiControllerDescription), Resources.ResourceManager, typeof(Resources));
+
+        private static readonly DiagnosticDescriptor Rule = new(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterSyntaxNodeAction(AnalyzeNode, cs.SyntaxKind.SimpleBaseType);
+            context.RegisterSyntaxNodeAction(AnalyzeNode, vb.SyntaxKind.InheritsStatement);
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var namedTypeSymbol = context.ContainingSymbol as INamedTypeSymbol;
+
+            if (namedTypeSymbol != null && namedTypeSymbol.BaseType != null
```

Comment:
I may be overlooking something here, but if you're getting the symbol for a C# SimpleBaseType syntax, I would think the type you need to look at is the type of that symbol itself rather than its BaseType.